test/processors.jl

```diff
@@ -40,23 +48,54 @@ end
             @test Dagger.iscompatible(proc, opts, unknown_func, us, 1, us, 2.0)
         end
     end
+    @testset "Function/argument compatability of DaggerKAProc" begin
+        unknown_func = () -> nothing
+        tp = DaggerKAProc(1)
+        op = get_parent(tp)
+        opts = ThunkOptions()
+        us = UnknownStruct()
+        for proc in (op, tp)
+            @test Dagger.iscompatible_func(proc, opts, unknown_func)
+            @test Dagger.iscompatible_arg(proc, opts, us)
+            @test Dagger.iscompatible(proc, opts, unknown_func, us, 1)
+        end
+    value(a) = a
+    @kernel function add(A, B, C)
+        I = @index(Global)
+        C[I] = A[I] + B[I]
+    end
+    A = delayed(value)(ones(512,512))
+    B = delayed(value)(3*ones(512,512))
+    C = delayed(value)(zeros(512,512))
+    S = delayed(add)(A,B,C)
```

Comment:
Yes, I had added that earlier but that will create a kernel object for the event. 
``delayed`` passes only functions. We need kernel functions so that dagger can create an object of itself from the kernel function.